### PR TITLE
Split up config cookbook

### PR DESCRIPTION
**Pull request by rosieyohannan**
# Description
* archived the config cookbook
* created the following how to guides from the content that was in the cookbook:
  * /docs/2.0/using-matrix-jobs/
  * /docs/2.0/using-branch-filters/
  * /docs/2.0/using-dynamic-configuration/
  * /docs/2.0/selecting-a-workflow-to-run-using-pipeline-parameters/
  * /docs/2.0/deploy-service-update-to-aws-ec2/
  * /docs/2.0/notify-a-slack-channel-of-a-paused-workflow/
* Added the how-to guides here (note one of the how to guides has changed title since I took these screenshots 🤪 ):
Screenshot 2022-06-01 at 15 05 51
Screenshot 2022-06-01 at 15 04 12

# Reasons
https://circleci.atlassian.net/browse/DOCSTEAM-283?atlOrigin=eyJpIjoiOTE3YTFiM2Q1OWE4NDgwOGJlZTYwYTc3MzdjYzE4NDkiLCJwIjoiaiJ9

# Content Checklist
Please follow our style when contributing to CircleCI docs. Our style guide is here: https://circleci.com/docs/style/.

Please take a moment to check through the following items when submitting your PR (this is just a guide so will not be relevant for all PRs) 😸:

- [x] Break up walls of text by adding paragraph breaks.
- [x] Consider if the content could benefit from more structure, such as lists or tables, to make it easier to consume.
- [x] Keep the title between 20 and 70 characters.
- [x] Consider whether the content would benefit from more subsections (h2-h6 headings) to make it easier to consume.
- [x] Check all headings h1-h6 are in sentence case (only first letter is capitalized).
- [x] Is there a "Next steps" section at the end of the page giving the reader a clear path to what to read next?
- [x] Include relevant backlinks to other CircleCI docs/pages.

### CODE REVIEW

**Review by w3bdesign — APPROVED** [review]

**Comment by devinmarieb** [comment]
@rosieyohannan I've looked over this and it looks good to me. With most pages we will need to update the file names to match the title of the article, but we can also do that later in a separate PR for a lot of this stuff.

**Comment by rosieyohannan** [comment]
> @rosieyohannan I've looked over this and it looks good to me. With most pages we will need to update the file names to match the title of the article, but we can also do that later in a separate PR for a lot of this stuff. If you're waiting on an approval I can do that now if it's ready to move out of draft.

Thanks @devinmarieb! Yes please to review - I have done a bit more work, especially on the orbs how-tos to try and make them flow more like the language guides. There is more work needed on the other how-to guides here but I would like to get this PR done really and I'll make separate tickets for further work required. 😅

**Comment by Plataformaalgoritmica** [comment]
Hello World.

File: jekyll/_cci2/deploy-service-update-to-aws-ec2.adoc [hunk 1]
```diff
@@ -121,7 +121,7 @@ jobs:
           task-definition-arn: '${TASK_DEFINITION_ARN}'
 ----
 
-This section illustrates how you can use the orb to install and configure the AWS CLI, retrieve the [task definition](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/task_definitions.html) that was previously deployed, and then _verify_ the revision has been deployed using the `verify-revision-is-deployed` command from the `AWS-ECS` orb.
+This section illustrates how you can use the orb to install and configure the AWS CLI, retrieve the link:https://docs.aws.amazon.com/AmazonECS/latest/developerguide/task_definitions.html[task definition] that was previously deployed, and then _verify_ the revision has been deployed using the `verify-revision-is-deployed` command from the `AWS-ECS` orb.
```
Comment (devinmarieb) [inline jekyll/_cci2/deploy-service-update-to-aws-ec2.adoc:124]: What is this syntax doing with the link:url[title]? I imagine it's adding the word "link:" before the link? Would there then need to be a space between "link:" and the definition name?

File: jekyll/_cci2/deploy-service-update-to-aws-ec2.adoc [hunk 2]
```diff
@@ -201,4 +201,4 @@ workflows:
 [#next-steps]
 == Next steps
 
-Find more detailed information in the CircleCI orb Registry for the CircleCI [AWS ECS](https://circleci.com/developer/orbs/orb/circleci/aws-ecs) and [AWS ECR](https://circleci.com/developer/orbs/orb/circleci/aws-ecr) orbs.
\ No newline at end of file
+* Find more detailed information in the CircleCI orb Registry for the CircleCI link:https://circleci.com/developer/orbs/orb/circleci/aws-ecs[AWS ECS] and link:https://circleci.com/developer/orbs/orb/circleci/aws-ecr[AWS ECR] orbs.
```
Comment (devinmarieb) [inline jekyll/_cci2/deploy-service-update-to-aws-ec2.adoc:204]: Same comment as above.

**Review by devinmarieb — APPROVED** [review]
Looks good! I just had a comment about the syntax on something. I'm unsure if it's a style thing or an asciidoc thing 🙃

File: jekyll/_cci2/deploy-service-update-to-aws-ec2.adoc (hunk 1 above)
Comment (rosieyohannan) [reply jekyll/_cci2/deploy-service-update-to-aws-ec2.adoc:124]: sorry for the confusion here - it is the same as just `url[custom link text]` ... I won't do it in future, I was trying it out as it looks easier to see in a code editor but its not necessary.

**Pull request merged**